src/Bicep.LangServer/Handlers/BicepDeploymentScopeRequestHandler.cs

```diff
@@ -0,0 +1,155 @@
+        private readonly ICompilationManager compilationManager;
+        private readonly IConfigurationManager configurationManager;
+        private readonly IFeatureProvider features;
+        private readonly IFileResolver fileResolver;
+        private readonly IModuleDispatcher moduleDispatcher;
+        private readonly INamespaceProvider namespaceProvider;
+
+        public BicepDeploymentScopeRequestHandler(
+            EmitterSettings emitterSettings,
+            ICompilationManager compilationManager,
+            IConfigurationManager configurationManager,
+            IFeatureProvider features,
+            IFileResolver fileResolver,
+            IModuleDispatcher moduleDispatcher,
+            INamespaceProvider namespaceProvider)
+        {
+            this.compilationManager = compilationManager;
+            this.configurationManager = configurationManager;
+            this.emitterSettings = emitterSettings;
+            this.features = features;
+            this.fileResolver = fileResolver;
+            this.moduleDispatcher = moduleDispatcher;
+            this.namespaceProvider = namespaceProvider;
+        }
+
+        public Task<BicepDeploymentScopeResponse> Handle(BicepDeploymentScopeParams request, CancellationToken cancellationToken)
+        {
+            var documentUri = request.TextDocument.Uri;
+
+            Compilation? compilation;
+
+            try
+            {
+                compilation = GetCompilation(documentUri);
+                var deploymentScope = GetDeploymentScope(compilation.GetEntrypointSemanticModel().TargetScope);
+
+                return Task.FromResult(new BicepDeploymentScopeResponse(deploymentScope, GetCompiledFile(compilation, documentUri), null));
+            }
+            catch (Exception exception)
+            {
+                return Task.FromResult(new BicepDeploymentScopeResponse(ResourceScope.None.ToString(), null, exception.Message));
+            }
+        }
+
+        private static string GetDeploymentScope(ResourceScope resourceScope)
+        {
+            switch (resourceScope)
+            {
+                case ResourceScope.ResourceGroup:
+                    return LanguageConstants.TargetScopeTypeResourceGroup;
+                case ResourceScope.Subscription:
+                    return LanguageConstants.TargetScopeTypeSubscription;
+                case ResourceScope.ManagementGroup:
+                    return LanguageConstants.TargetScopeTypeManagementGroup;
+                case ResourceScope.Tenant:
+                    return LanguageConstants.TargetScopeTypeTenant;
+                default:
+                    return resourceScope.ToString();
+            }
+        }
+
+        private string GetCompiledFile(Compilation compilation, DocumentUri documentUri)
+        {
+            var fileUri = documentUri.ToUri();
+
+            KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile = compilation.GetAllDiagnosticsByBicepFile()
```

Comment:
Not related to this PR, but I just realized we can simplify the return type of `GetAllDiagnosticsByBicepFile` to `ILookup<BicepFile, IDiagnostic>`. Created https://github.com/Azure/bicep/issues/6179 to track it.

Reply:
Thank you!